### DISCUSSION TITLE
Allow profiler to be used in an editor context

### DISCUSSION THREAD

**Discussion by ryevdokimov** [godotengine/godot-proposals#10851] [Editor]
### Describe the project you are working on

Simulation software that is entirely ran in the editor

### Describe the problem or limitation you are having in your project

Not being able to profile the editor itself

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Allow profiling the editor, to diagnose issues with tool scripts, editor plugins, engine modifications, etc.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

The same way as the current profiler, but using the editor itself as the context 

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Not a few.

### Is there a reason why this should be core and not an add-on in the asset library?

It would be useful to anyone developing more complex addons.

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.
